02-microarray/differential-expression_microarray_01_2-groups.Rmd

```diff
@@ -238,43 +238,63 @@ all.equal(colnames(df), metadata$geo_accession)
 ## Set up design matrix
 
 `limma` needs a numeric design matrix to signify which are CREB and control samples.
-Here we are using the treatments supplied in the metadata to create a design matrix where the "none" samples are assigned `0` and the "amputated" samples are assigned `1`.
-Note that the metadata variables that signify the treatment groups might be different across datasets and might not always be underneath the category.
+Here we are using the treatments described in the metadata table in the `genotype/variation` column to create a design matrix where the "control" samples are assigned `0` and the "overexpressing the human CREB" samples are assigned `1`.
+Note that the metadata columns that signify the treatment groups might be different across datasets, and will almost certainly have different contents.
 
-The `genotype/variation` column contains group information we will be using for differential expression.
-But the `/` it contains in its column name makes it more annoying to access.  
+While the `genotype/variation` column contains the group information we will be using for differential expression, the `/` it contains in its column name makes it more annoying to access.  
 Accessing variable that have names with special characters like `/`, or spaces, require extra work-arounds to ignore R's normal interpretations of these characters.
+Here we will rename it as just `genotype` to make our lives later much easier.
+
+We will also recode the contents of the column, as `overexpressing the human CREB"` is a bit of an unruly name. 
+To do this, we will use a function from the `forcats` package, simplifying `"overexpressing the human CREB"` to just `CREB`.
+
 
 ```{r}
+# These renaming steps will not be the same (or might not be needed at all)
+# with a different dataset
 metadata <- metadata %>%
-  dplyr::rename("genotype" = `genotype/variation`) # This step will not be the same (or might not be needed at all) with a different dataset
+  # rename the column
+  dplyr::rename("genotype" = `genotype/variation`) %>%
+  # change the name of the CREB genotype (and make genotype a factor)
+  dplyr::mutate(
+    genotype = forcats::fct_recode(genotype,
+      CREB = "overexpressing the human CREB"
```

Comment:
Might want a comment here to explain this particular line since its the "meat" of this step.